algoliasearch-core/src/main/java/com/algolia/search/models/HttpRequest.java

```diff
@@ -69,6 +80,27 @@ public HttpRequest setTimeout(int timeout) {
     return this;
   }
 
+  public CompressionType getCompressionType() {
+    return compressionType;
+  }
+
+  public HttpRequest setCompressionType(CompressionType compressionType) {
+    this.compressionType = compressionType;
+    return this;
+  }
+
+  /**
+   * Tells if any compression can be enabled for a request or not. Compression is enabled only for
+   * POST/PUT methods on the Search API (not on Analytics and Insights).
+   */
+  public boolean canCompress() {
+    if (this.compressionType == null || this.method == null) {
+      return false;
+    } else
+      return this.compressionType.equals(CompressionType.GZIP)
+          && (this.method.equals(HttpMethod.POST) || this.method.equals(HttpMethod.PUT));
+  }
```

Comment:
All these nested `if`/`else` could be simplified a bit such as:

```
var isMethodValid = method.equals.(POST) || method.equals(PUT);
var isCompressionEnabled = ! compression.equals(NONE)

return isMethodValid && isCompressionEnabled
```

Also, you mentioned in the comments that should never return true for Analytics and Insights API so you may want to introduce a third boolean here to check that the API kind is the Search API and to add it to the return call.

WDYT?

Reply:
> All these nested `if`/`else` could be simplified a bit such as:

Sure let's simplify that 👍 

> Also, you mentioned in the comments that should never return true for Analytics and Insights API so you may want to introduce a third boolean here to check that the API kind is the Search API and to add it to the return call.

Yes, to have a more robust code we should introduce another test. Here is my proposal:

The `HttpRequest` object is taking the `compression` property from the configuration in `HttpTransport`. Let's add an additional control in`configuration.getCompressionType()` to test whether or not it's an Analytic/Insight config instance and return false for both of them. 

WDYT?

Reply:
Always all in for more tests 💯

Reply:
@aseure I fixed the two comments above with 155ee4.

`if/else` statement are now simplified. Regarding extra check for the compression type, I found a way to have `setCompressionType()` only avaible on `SearchConfig`. Now there is no way for a user to set a `compressionType` for a config that doesn't support it, such as `Analytics` or `Insights.`